tests/test.collector.js

```diff
@@ -65,4 +65,40 @@ describe('Collector', function() {
     assert.equal(collection.notices.length, 0);
   });
 
+  it('should filter message by filename if config.scanFile is defined', () => {
+    var collection = new Collector({
+      scanFile: ['test.js', 'no-match-file.js'],
+    });
+
+    assert.equal(collection.length, 0);
+    collection.addError({
+      ...fakeMessageData,
+      // not file name
+    });
+    assert.equal(collection.length, 1);
+    assert.equal(collection.errors[0].type, 'error');
```

Comment:
You shouldn't need to test the type here as things in the `errors` array are always `'error'` type (and the collection class tests that).